backend/core/domains/payments/services/gateway_service.py

```
@@ -108,44 +158,154 @@ def _process_stripe_payment(payment_id, payment_data, user):
-        elif payment.payment_method and payment.payment_method.gateway:
+        logger.info(f"Getting Stripe gateway with ID: {gateway_id} (type: {type(gateway_id)})")
+        
+        gateway = None
+        if gateway_id and isinstance(gateway_id, (int, str)) and str(gateway_id).isdigit():
+            try:
+                # Ensure gateway_id is integer for database lookup
+                gateway_id_int = int(gateway_id)
+                gateway = PaymentGateway.objects.get(id=gateway_id_int, code='stripe')
+                logger.info(f"Found gateway by ID: {gateway.name}")
+            except PaymentGateway.DoesNotExist:
+                logger.error(f"Gateway with ID {gateway_id} not found")
```

Ruff (0.13.3)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)